### PR TITLE
LogicApp API Connection

### PR DESCRIPTION
**Pull request by uday31in**
**Proposed changes**:

- updating storage connection to ensure SAS key can be generated
- creating access policies for api connection

### CODE REVIEW

**Comment by uday31in** [comment]
Screenshot 2024-05-03 at 22 58 01

@marvinbuss  - to add access policies

**Comment by uday31in** [comment]
image

File: infrastructure/terraform/modules/logicapp/logicapp_apiconnections.tf [hunk 1]
```diff
@@ -39,3 +39,54 @@ resource "azapi_resource" "api_connection_arm" {
...
+    {
+      "connectionName" = {
+        value = "${var.logic_app_name}-api-${each.key}"
+      },
+      "principalId" = {
+        value = "${azurerm_logic_app_standard.logic_app_standard.identity[0].principal_id}"
+      }
+    }
+  )
+}
```
Comment (marvinbuss) [inline infrastructure/terraform/modules/logicapp/logicapp_apiconnections.tf:92]: I have done the same using azapi. If you don't mind let's use azapi as ARM templates usually create some challenges from a plan perspective.

File: infrastructure/terraform/modules/storageaccount/output.tf [hunk 2]
```diff
@@ -15,3 +15,9 @@ output "storage_account_primary_blob_endpoint" {
   description = "Specifies the primary blob endpoint of the storage account"
   sensitive   = false
 }
+
+output "storage_account_primary_connection_string" {
+  value       = azurerm_storage_account.storage.primary_connection_string
+  description = "Specifies the primary blob endpoint of the storage account"
+  sensitive   = true
+}
```
Comment (marvinbuss) [inline infrastructure/terraform/modules/storageaccount/output.tf:23]: I worked on a solution to generate SAS using HTTP. Also, this does not use account level SAS keys. I consider this more secure. What are your thoughts? Let's sync on that.

**Review by marvinbuss — CHANGES_REQUESTED** [review]
Please review comments.

**Comment by uday31in** [comment]
Please merge your changes and close this PR

File: infrastructure/terraform/modules/logicapp/logicapp_apiconnections.tf (hunk 1 above)
Comment (marvinbuss) [reply infrastructure/terraform/modules/logicapp/logicapp_apiconnections.tf:92]: This PR now includes the `azapi` resource: https://github.com/Azure/ai-hub/pull/166

**Comment by marvinbuss** [comment]
> Please merge your changes and close this PR

Will close this once we merged the other PR.

**Comment by marvinbuss** [comment]
Superseded by https://github.com/Azure/ai-hub/pull/166

**Pull request closed without merging**